### PR TITLE
🌱  Update golangci-lint to v1.51.0

### PR DESCRIPTION
**Pull request by killianmuldoon**
Release notes: https://github.com/golangci/golangci-lint/releases/tag/v1.51.0

### CODE REVIEW

File: util/container/image.go [hunk 1]
```diff
@@ -17,8 +17,7 @@ limitations under the License.
 // Package container implements container utility functionality.
 package container
 
-import (
-	//  Import the crypto sha256 algorithm for the docker image parser to work
+import ( //  Import the crypto sha256 algorithm for the docker image parser to work
```
Comment (killianmuldoon) [inline util/container/image.go:20]: Looks like the gci linter has some new opinions about comment placement - it looks worse to me, but I'm fine with merging this as it won't impact in many places.

Comment (sbueringer) [reply util/container/image.go]: Nice. I prefer the new variant. I think it's the mainstream variant (IIRC)

Comment (killianmuldoon) [reply util/container/image.go]: Yeah - the current version is nice. The outdated version above was the result from `make lint-fix` and it is ... not nice

**Comment by killianmuldoon** [comment]
>interesting to test the new ginkgolinter in a follow-up (I didn't take a closer look what kind of changes it would enforce, so we definitely would have to take a look first before we decide)

I couldn't get this to run quickly so I decided to do a quick update - it was crashing for me :thinking:

**Comment by oscr** [comment]
> I couldn't get this to run quickly so I decided to do a quick update - it was crashing for me thinking

I was actually experimenting with a similar pr as this one, but ginkgolinter was also crashing for me :grin:

**Comment by sbueringer** [comment]
Should we track it in an issue for now and then re-evaluate once those issues are resolved? (we should probably report them when we have time)

**Pull request merged**